Cant install on 64bit Linux with node 12.7.0

Pre-built binaries for node 12.7.0 are not found, and it then fails to compiled on 64 bit linux. Compiler output as follow:

`  TOUCH Release/obj.target/action_before_build.stamp
  ACTION binding_gyp_action_before_build2_target_run_protoc_glyphs Release/obj/gen/glyphs.pb.cc
  TOUCH Release/obj.target/action_before_build2.stamp
  CXX(target) Release/obj.target/fontnik/src/node_fontnik.o
In file included from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan_new.h:189:0,
                 from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h:203,
                 from ../src/glyphs.hpp:5,
                 from ../src/node_fontnik.cpp:2:
/home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan_implementation_12_inl.h: In static member function 'static Nan::imp::FactoryBase<v8::Function>::return_t Nan::imp::Factory<v8::Function>::New(Nan::FunctionCallback, v8::Local<v8::Value>)':
/home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan_implementation_12_inl.h:105:32: error: no matching function for call to 'v8::Function::New(v8::Isolate*&, void (&)(const v8::FunctionCallbackInfo<v8::Value>&), v8::Local<v8::Object>&)'
                           , obj));
                                ^
In file included from /home/tom/.node-gyp/12.7.0/include/node/node.h:63:0,
                 from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h:51,
                 from ../src/glyphs.hpp:5,
                 from ../src/node_fontnik.cpp:2:
/home/tom/.node-gyp/12.7.0/include/node/v8.h:4132:31: note: candidate: static v8::MaybeLocal<v8::Function> v8::Function::New(v8::Local<v8::Context>, v8::FunctionCallback, v8::Local<v8::Value>, int, v8::ConstructorBehavior, v8::SideEffectType)
   static MaybeLocal<Function> New(
                               ^
/home/tom/.node-gyp/12.7.0/include/node/v8.h:4132:31: note:   no known conversion for argument 1 from 'v8::Isolate*' to 'v8::Local<v8::Context>'
In file included from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan_new.h:189:0,
                 from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h:203,
                 from ../src/glyphs.hpp:5,
                 from ../src/node_fontnik.cpp:2:
/home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan_implementation_12_inl.h: In static member function 'static Nan::imp::FactoryBase<v8::StringObject>::return_t Nan::imp::Factory<v8::StringObject>::New(v8::Local<v8::String>)':
/home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan_implementation_12_inl.h:337:37: error: no matching function for call to 'v8::StringObject::New(v8::Local<v8::String>&)'
   return v8::StringObject::New(value).As<v8::StringObject>();
                                     ^
In file included from /home/tom/.node-gyp/12.7.0/include/node/node.h:63:0,
                 from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h:51,
                 from ../src/glyphs.hpp:5,
                 from ../src/node_fontnik.cpp:2:
/home/tom/.node-gyp/12.7.0/include/node/v8.h:5388:23: note: candidate: static v8::Local<v8::Value> v8::StringObject::New(v8::Isolate*, v8::Local<v8::String>)
   static Local<Value> New(Isolate* isolate, Local<String> value);
                       ^
/home/tom/.node-gyp/12.7.0/include/node/v8.h:5388:23: note:   candidate expects 2 arguments, 1 provided
In file included from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan_new.h:189:0,
                 from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h:203,
                 from ../src/glyphs.hpp:5,
                 from ../src/node_fontnik.cpp:2:
/home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan_implementation_12_inl.h:337:58: error: expected primary-expression before '>' token
   return v8::StringObject::New(value).As<v8::StringObject>();
                                                          ^
/home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan_implementation_12_inl.h:337:60: error: expected primary-expression before ')' token
   return v8::StringObject::New(value).As<v8::StringObject>();
                                                            ^
In file included from ../src/glyphs.hpp:5:0,
                 from ../src/node_fontnik.cpp:2:
/home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h: In constructor 'Nan::Utf8String::Utf8String(v8::Local<v8::Value>)':
/home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h:1034:53: error: no matching function for call to 'v8::Value::ToString()'
       v8::Local<v8::String> string = from->ToString();
                                                     ^
In file included from /home/tom/.node-gyp/12.7.0/include/node/node.h:63:0,
                 from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h:51,
                 from ../src/glyphs.hpp:5,
                 from ../src/node_fontnik.cpp:2:
/home/tom/.node-gyp/12.7.0/include/node/v8.h:2535:44: note: candidate: v8::MaybeLocal<v8::String> v8::Value::ToString(v8::Local<v8::Context>) const
   V8_WARN_UNUSED_RESULT MaybeLocal<String> ToString(
                                            ^
/home/tom/.node-gyp/12.7.0/include/node/v8.h:2535:44: note:   candidate expects 1 argument, 0 provided
In file included from /home/tom/.node-gyp/12.7.0/include/node/v8-internal.h:14:0,
                 from /home/tom/.node-gyp/12.7.0/include/node/v8.h:25,
                 from /home/tom/.node-gyp/12.7.0/include/node/node.h:63,
                 from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h:51,
                 from ../src/glyphs.hpp:5,
                 from ../src/node_fontnik.cpp:2:
/home/tom/.node-gyp/12.7.0/include/node/v8.h:2550:3: note: candidate: v8::Local<v8::String> v8::Value::ToString(v8::Isolate*) const
   V8_DEPRECATED("Use maybe version",
   ^
/home/tom/.node-gyp/12.7.0/include/node/v8.h:2550:3: note:   candidate expects 1 argument, 0 provided
In file included from ../src/glyphs.hpp:5:0,
                 from ../src/node_fontnik.cpp:2:
/home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h:1044:74: error: no matching function for call to 'v8::String::WriteUtf8(char*&, int, int, const int&)'
         length_ = string->WriteUtf8(str_, static_cast<int>(len), 0, flags);
                                                                          ^
In file included from /home/tom/.node-gyp/12.7.0/include/node/node.h:63:0,
                 from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h:51,
                 from ../src/glyphs.hpp:5,
                 from ../src/node_fontnik.cpp:2:
/home/tom/.node-gyp/12.7.0/include/node/v8.h:2745:7: note: candidate: int v8::String::WriteUtf8(v8::Isolate*, char*, int, int*, int) const
   int WriteUtf8(Isolate* isolate, char* buffer, int length = -1,
       ^
/home/tom/.node-gyp/12.7.0/include/node/v8.h:2745:7: note:   no known conversion for argument 1 from 'char*' to 'v8::Isolate*'
../src/node_fontnik.cpp: In function 'void node_fontnik::RegisterModule(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)':
../src/node_fontnik.cpp:8:102: error: no matching function for call to 'v8::FunctionTemplate::GetFunction()'
     target->Set(Nan::New("load").ToLocalChecked(), Nan::New<v8::FunctionTemplate>(Load)->GetFunction());
                                                                                                      ^
In file included from /home/tom/.node-gyp/12.7.0/include/node/node.h:63:0,
                 from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h:51,
                 from ../src/glyphs.hpp:5,
                 from ../src/node_fontnik.cpp:2:
/home/tom/.node-gyp/12.7.0/include/node/v8.h:5955:46: note: candidate: v8::MaybeLocal<v8::Function> v8::FunctionTemplate::GetFunction(v8::Local<v8::Context>)
   V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
                                              ^
/home/tom/.node-gyp/12.7.0/include/node/v8.h:5955:46: note:   candidate expects 1 argument, 0 provided
../src/node_fontnik.cpp:9:104: error: no matching function for call to 'v8::FunctionTemplate::GetFunction()'
    target->Set(Nan::New("range").ToLocalChecked(), Nan::New<v8::FunctionTemplate>(Range)->GetFunction());
                                                                                                       ^
In file included from /home/tom/.node-gyp/12.7.0/include/node/node.h:63:0,
                 from /home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/node_modules/nan/nan.h:51,
                 from ../src/glyphs.hpp:5,
                 from ../src/node_fontnik.cpp:2:
/home/tom/.node-gyp/12.7.0/include/node/v8.h:5955:46: note: candidate: v8::MaybeLocal<v8::Function> v8::FunctionTemplate::GetFunction(v8::Local<v8::Context>)
   V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
                                              ^
/home/tom/.node-gyp/12.7.0/include/node/v8.h:5955:46: note:   candidate expects 1 argument, 0 provided
fontnik.target.mk:154: recipe for target 'Release/obj.target/fontnik/src/node_fontnik.o' failed
make: *** [Release/obj.target/fontnik/src/node_fontnik.o] Error 1
make: Leaving directory '/home/linuxbrew/.linuxbrew/lib/node_modules/fontnik/build'
`

--- a/.travis.yml
+++ b/.travis.yml
@@ -39,22 +39,6 @@ script:
 # run your tests and build binaries
 matrix:
   include:
-    # linux publishable node v4/release
-    - os: linux
-      env: BUILDTYPE=release
-      node_js: 4
-    # linux publishable node v4/debug
-    - os: linux
-      env: BUILDTYPE=debug
-      node_js: 4
-    # linux publishable node v6
-    - os: linux
-      env: BUILDTYPE=release
-      node_js: 6
-    # linux publishable node v6/debug
-    - os: linux
-      env: BUILDTYPE=debug
-      node_js: 6
     # linux publishable node v8
     - os: linux
       env: BUILDTYPE=release
@@ -63,38 +47,41 @@ matrix:
     - os: linux
       env: BUILDTYPE=debug
       node_js: 8
-    # linux publishable node v8
+    # linux publishable node v10
     - os: linux
       env: BUILDTYPE=release
       node_js: 10
-    # linux publishable node v8/debug
+    # linux publishable node v10/debug
     - os: linux
       env: BUILDTYPE=debug
       node_js: 10
-    # osx publishable node v4
-    - os: osx
-      osx_image: xcode8.2
+    # linux publishable node v10
+    - os: linux
       env: BUILDTYPE=release
-      node_js: 4
-    # osx publishable node v6
-    - os: osx
-      osx_image: xcode8.2
-      env: BUILDTYPE=release
-      node_js: 6
-    # osx publishable node v6
+      node_js: 12
+    # linux publishable node v10/debug
+    - os: linux
+      env: BUILDTYPE=debug
+      node_js: 12
+    # osx publishable node v8
     - os: osx
       osx_image: xcode8.2
       env: BUILDTYPE=release
       node_js: 8
-    # osx publishable node v6
+    # osx publishable node v10
     - os: osx
       osx_image: xcode8.2
       env: BUILDTYPE=release
       node_js: 10
-    # Sanitizer build node v4/Debug
+    # osx publishable node v12
+    - os: osx
+      osx_image: xcode8.2
+      env: BUILDTYPE=release
+      node_js: 12
+    # Sanitizer build node v10/Debug
     - os: linux
       env: BUILDTYPE=debug TOOLSET=asan
-      node_js: 4
+      node_js: 10
       sudo: required
       # Overrides `install` to set up custom asan flags
       install:
@@ -119,7 +106,7 @@ matrix:
     # g++ build (default builds all use clang++)
     - os: linux
       env: BUILDTYPE=debug CXX="g++-6" CC="gcc-6"
-      node_js: 4
+      node_js: 10
       addons:
         apt:
           sources:
@@ -136,7 +123,7 @@ matrix:
     # Coverage build
     - os: linux
       env: BUILDTYPE=debug CXXFLAGS="--coverage" LDFLAGS="--coverage"
-      node_js: 4
+      node_js: 10
       # Overrides `script` to publish coverage data to codecov
       before_script:
         - npm test
@@ -164,7 +151,7 @@ matrix:
     # Clang tidy build
     - os: linux
       env: CLANG_TIDY
-      node_js: 4
+      node_js: 10
       # Overrides `install` to avoid initializing clang toolchain
       install:
         # First run the clang-tidy target

--- a/.travis.yml
+++ b/.travis.yml
@@ -105,9 +105,9 @@ matrix:
         - true
     # g++ build (default builds all use clang++)
     - os: linux
-      # Note: -std=gnu++14 is needed to workaround:
+      # Note: -fext-numeric-literals is needed to workaround gcc bug:
       # boost/math/constants/constants.hpp:269:3: error: unable to find numeric literal operator 'operatorQ'
-      env: BUILDTYPE=debug CXX="g++-6" CC="gcc-6" CXXFLAGS="-std=gnu++14"
+      env: BUILDTYPE=debug CXX="g++-6" CC="gcc-6" CXXFLAGS="-fext-numeric-literals"
       node_js: 10
       addons:
         apt:

--- a/.travis.yml
+++ b/.travis.yml
@@ -105,7 +105,9 @@ matrix:
         - true
     # g++ build (default builds all use clang++)
     - os: linux
-      env: BUILDTYPE=debug CXX="g++-6" CC="gcc-6"
+      # Note: -std=gnu++14 is needed to workaround:
+      # boost/math/constants/constants.hpp:269:3: error: unable to find numeric literal operator 'operatorQ'
+      env: BUILDTYPE=debug CXX="g++-6" CC="gcc-6" CXXFLAGS="-std=gnu++14"
       node_js: 10
       addons:
         apt:

--- a/package.json
+++ b/package.json
@@ -25,7 +25,7 @@
   "dependencies": {
     "glob": "^7.1.1",
     "minimist": "^0.2.0",
-    "nan": "~2.10.0",
+    "nan": "^2.14.0",
     "node-pre-gyp": "^0.12.0",
     "queue-async": "^1.0.7"
   },

--- a/scripts/install_deps.sh
+++ b/scripts/install_deps.sh
@@ -12,7 +12,7 @@ function install() {
 ./scripts/setup.sh --config local.env
 source local.env
 
-install boost 1.63.0
+install boost 1.67.0
 install freetype 2.7.1
 install protozero 1.6.8
 install sdf-glyph-foundry 0.1.1

--- a/scripts/setup.sh
+++ b/scripts/setup.sh
@@ -85,6 +85,7 @@ function run() {
     echo "leak:v8::internal" >> ${SUPPRESSION_FILE}
     echo "leak:node::CreateEnvironment" >> ${SUPPRESSION_FILE}
     echo "leak:node::Init" >> ${SUPPRESSION_FILE}
+    echo "leak:node::Buffer::Copy" >> ${SUPPRESSION_FILE}
     echo "export ASAN_SYMBOLIZER_PATH=${llvm_toolchain_dir}/bin/llvm-symbolizer" >> ${config}
     echo "export MSAN_SYMBOLIZER_PATH=${llvm_toolchain_dir}/bin/llvm-symbolizer" >> ${config}
     echo "export UBSAN_OPTIONS=print_stacktrace=1" >> ${config}

--- a/src/glyphs.cpp
+++ b/src/glyphs.cpp
@@ -361,10 +361,10 @@ void AfterComposite(uv_work_t* req) {
     Nan::HandleScope scope;
 
     CompositeBaton* baton = static_cast<CompositeBaton*>(req->data);
-
+    Nan::AsyncResource async_resource(__func__);
     if (!baton->error_name.empty()) {
         v8::Local<v8::Value> argv[1] = {Nan::Error(baton->error_name.c_str())};
-        Nan::MakeCallback(Nan::GetCurrentContext()->Global(), Nan::New(baton->callback), 1, argv);
+        async_resource.runInAsyncScope(Nan::GetCurrentContext()->Global(), Nan::New(baton->callback), 1, argv);
     } else {
         std::string& fontstack_message = *baton->message.get();
         const auto argc = 2u;
@@ -377,7 +377,7 @@ void AfterComposite(uv_work_t* req) {
                            },
                            baton->message.release())
                 .ToLocalChecked()};
-        Nan::MakeCallback(Nan::GetCurrentContext()->Global(), Nan::New(baton->callback), 2, argv);
+        async_resource.runInAsyncScope(Nan::GetCurrentContext()->Global(), Nan::New(baton->callback), 2, argv);
     }
 
     delete baton;
@@ -468,10 +468,10 @@ void AfterLoad(uv_work_t* req) {
     Nan::HandleScope scope;
 
     LoadBaton* baton = static_cast<LoadBaton*>(req->data);
-
+    Nan::AsyncResource async_resource(__func__);
     if (!baton->error_name.empty()) {
         v8::Local<v8::Value> argv[1] = {Nan::Error(baton->error_name.c_str())};
-        Nan::MakeCallback(Nan::GetCurrentContext()->Global(), Nan::New(baton->callback), 1, argv);
+        async_resource.runInAsyncScope(Nan::GetCurrentContext()->Global(), Nan::New(baton->callback), 1, argv);
     } else {
         v8::Local<v8::Array> js_faces = Nan::New<v8::Array>(baton->faces.size());
         unsigned idx = 0;
@@ -488,7 +488,7 @@ void AfterLoad(uv_work_t* req) {
             js_faces->Set(idx++, js_face);
         }
         v8::Local<v8::Value> argv[2] = {Nan::Null(), js_faces};
-        Nan::MakeCallback(Nan::GetCurrentContext()->Global(), Nan::New(baton->callback), 2, argv);
+        async_resource.runInAsyncScope(Nan::GetCurrentContext()->Global(), Nan::New(baton->callback), 2, argv);
     }
     delete baton;
 }
@@ -605,13 +605,13 @@ void AfterRange(uv_work_t* req) {
     Nan::HandleScope scope;
 
     RangeBaton* baton = static_cast<RangeBaton*>(req->data);
-
+    Nan::AsyncResource async_resource(__func__);
     if (!baton->error_name.empty()) {
         v8::Local<v8::Value> argv[1] = {Nan::Error(baton->error_name.c_str())};
-        Nan::MakeCallback(Nan::GetCurrentContext()->Global(), Nan::New(baton->callback), 1, argv);
+        async_resource.runInAsyncScope(Nan::GetCurrentContext()->Global(), Nan::New(baton->callback), 1, argv);
     } else {
         v8::Local<v8::Value> argv[2] = {Nan::Null(), Nan::CopyBuffer(baton->message.data(), static_cast<std::uint32_t>(baton->message.size())).ToLocalChecked()};
-        Nan::MakeCallback(Nan::GetCurrentContext()->Global(), Nan::New(baton->callback), 2, argv);
+        async_resource.runInAsyncScope(Nan::GetCurrentContext()->Global(), Nan::New(baton->callback), 2, argv);
     }
 
     delete baton;

--- a/src/glyphs.cpp
+++ b/src/glyphs.cpp
@@ -155,7 +155,7 @@ NAN_METHOD(Load) {
     if (!info[0]->IsObject()) {
         return Nan::ThrowTypeError("First argument must be a font buffer");
     }
-    v8::Local<v8::Object> obj = info[0]->ToObject();
+    v8::Local<v8::Object> obj = info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
     if (obj->IsNull() || obj->IsUndefined() || !node::Buffer::HasInstance(obj)) {
         return Nan::ThrowTypeError("First argument must be a font buffer");
     }
@@ -179,7 +179,7 @@ NAN_METHOD(Range) {
     if (!font_buffer->IsObject()) {
         return Nan::ThrowTypeError("Font buffer is not an object");
     }
-    v8::Local<v8::Object> obj = font_buffer->ToObject();
+    v8::Local<v8::Object> obj = font_buffer->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
     v8::Local<v8::Value> start = options->Get(Nan::New<v8::String>("start").ToLocalChecked());
     v8::Local<v8::Value> end = options->Get(Nan::New<v8::String>("end").ToLocalChecked());
 
@@ -187,15 +187,15 @@ NAN_METHOD(Range) {
         return Nan::ThrowTypeError("First argument must be a font buffer");
     }
 
-    if (!start->IsNumber() || start->IntegerValue() < 0) {
+    if (!start->IsNumber() || Nan::To<std::int32_t>(start).FromJust() < 0) {
         return Nan::ThrowTypeError("option `start` must be a number from 0-65535");
     }
 
-    if (!end->IsNumber() || end->IntegerValue() > 65535) {
+    if (!end->IsNumber() || Nan::To<std::int32_t>(end).FromJust() > 65535) {
         return Nan::ThrowTypeError("option `end` must be a number from 0-65535");
     }
 
-    if (end->IntegerValue() < start->IntegerValue()) {
+    if (Nan::To<std::int32_t>(end).FromJust() < Nan::To<std::int32_t>(start).FromJust()) {
         return Nan::ThrowTypeError("`start` must be less than or equal to `end`");
     }
 
@@ -205,8 +205,8 @@ NAN_METHOD(Range) {
 
     RangeBaton* baton = new RangeBaton(obj,
                                        info[1],
-                                       start->Uint32Value(),
-                                       end->Uint32Value());
+                                       Nan::To<std::uint32_t>(start).FromJust(),
+                                       Nan::To<std::uint32_t>(end).FromJust());
     uv_queue_work(uv_default_loop(), &baton->request, RangeAsync, reinterpret_cast<uv_after_work_cb>(AfterRange));
 }
 
@@ -244,14 +244,12 @@ NAN_METHOD(Composite) {
 
     CompositeBaton* baton = new CompositeBaton(num_glyphs, callback);
 
-    v8::Isolate* isolate = v8::Isolate::GetCurrent();
-
     for (unsigned t = 0; t < num_glyphs; ++t) {
         v8::Local<v8::Value> buf_val = glyphs->Get(t);
         if (buf_val->IsNull() || buf_val->IsUndefined()) {
             return utils::CallbackError("buffer value in 'glyphs' array item is null or undefined", callback);
         }
-        v8::MaybeLocal<v8::Object> maybe_buffer = buf_val->ToObject(isolate->GetCurrentContext());
+        v8::MaybeLocal<v8::Object> maybe_buffer = buf_val->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
         if (maybe_buffer.IsEmpty()) {
             return utils::CallbackError("buffer value in 'glyphs' array is empty", callback);
         }

--- a/src/node_fontnik.cpp
+++ b/src/node_fontnik.cpp
@@ -5,9 +5,9 @@
 namespace node_fontnik {
 
 static void init(v8::Local<v8::Object> target) {
-    Nan::SetMethod(target,"load",Load);
-    Nan::SetMethod(target,"range",Range);
-    Nan::SetMethod(target,"composite",Composite);
+    Nan::SetMethod(target, "load", Load);
+    Nan::SetMethod(target, "range", Range);
+    Nan::SetMethod(target, "composite", Composite);
 }
 
 // We mark this NOLINT to avoid the clang-tidy checks

--- a/src/node_fontnik.cpp
+++ b/src/node_fontnik.cpp
@@ -4,15 +4,15 @@
 
 namespace node_fontnik {
 
-NAN_MODULE_INIT(RegisterModule) {
-    target->Set(Nan::New("load").ToLocalChecked(), Nan::New<v8::FunctionTemplate>(Load)->GetFunction());
-    target->Set(Nan::New("range").ToLocalChecked(), Nan::New<v8::FunctionTemplate>(Range)->GetFunction());
-    target->Set(Nan::New("composite").ToLocalChecked(), Nan::New<v8::FunctionTemplate>(Composite)->GetFunction());
+static void init(v8::Local<v8::Object> target) {
+    Nan::SetMethod(target,"load",Load);
+    Nan::SetMethod(target,"range",Range);
+    Nan::SetMethod(target,"composite",Composite);
 }
 
 // We mark this NOLINT to avoid the clang-tidy checks
 // warning about code inside nodejs that we don't control and can't
 // directly change to avoid the warning.
-NODE_MODULE(fontnik, RegisterModule) // NOLINT
+NODE_MODULE(fontnik, init) // NOLINT
 
 } // namespace node_fontnik

